### PR TITLE
Add docstrings to transformers components

### PR DESCRIPTION
**Pull request by dakshvar22**
**Proposed changes**:
- Related to https://github.com/RasaHQ/rasa/issues/5319

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/utils/hugging_face/hf_transformers.py [hunk 1]
```diff
@@ -147,6 +185,21 @@ def _post_process_sequence_embeddings(
...
+
+        Many language models add a special char in front of (some) words and split words into
+        sub-words. To ensure the entity start and end values matches the token values,
+        tokenize the text first using the whitespace tokenizer. If individual tokens
+        are split up into multiple tokens, we make sure that the start and end value
+        of the first and last respective tokens stay the same.
+
+        Args:
+            message: Single message object to be processed.
+            attribute: Property of message to be processed, one of ``INTENT`` or ``RESPONSE``.
```
Comment (tabergma) [inline rasa/nlu/utils/hugging_face/hf_transformers.py]: > one of ``INTENT`` or ``RESPONSE``.

Are you sure? Not `one of ``TEXT`` and ``RESPONSE``.`?

File: rasa/nlu/utils/hugging_face/hf_transformers.py [hunk 2]
```diff
@@ -173,6 +226,15 @@ def _tokenize_example(
     def _get_token_ids_for_batch(
         self, batch_examples: List[Message], attribute: Text
     ) -> Tuple[List[List[Token]], List[List[int]]]:
+        """Compute token ids and token strings for each example in batch.
+
+        Args:
+            batch_examples: Batch of message objects for which tokens need to be computed.
+            attribute: Property of message to be processed, one of ``INTENT`` or ``RESPONSE``.
```
Comment (tabergma) [inline rasa/nlu/utils/hugging_face/hf_transformers.py]: same

File: rasa/nlu/utils/hugging_face/hf_transformers.py [hunk 3]
```diff
@@ -287,6 +398,15 @@ def _get_model_features_for_batch(
     def _get_docs_for_batch(
         self, batch_examples: List[Message], attribute: Text
     ) -> List[Dict[Text, Any]]:
+        """Compute language model docs for all examples in the batch.
+
+        Args:
+            batch_examples: Batch of message objects for which language model docs need to be computed.
+            attribute: Property of message to be processed, one of ``INTENT`` or ``RESPONSE``.
```
Comment (tabergma) [inline rasa/nlu/utils/hugging_face/hf_transformers.py]: same

File: rasa/nlu/utils/hugging_face/transformers_pre_post_processors.py [hunk 4]
```diff
@@ -5,7 +5,15 @@
 def cleanup_tokens(
     token_ids_string: List[Tuple[int, Text]], delimiter: Text
 ) -> Tuple[List[int], List[Text]]:
-    """Utility method to apply specific delimiter based cleanup on list of tokens"""
+    """Utility method to apply delimiter based cleanup on list of tokens.
+
+    Args:
+        token_ids_string: List of tuples with each tuple containing (token id, token string).
```
Comment (tabergma) [inline rasa/nlu/utils/hugging_face/transformers_pre_post_processors.py]: Just for my understanding: token id = position of token in that message? Or token id = id in vocab?

File: rasa/nlu/utils/hugging_face/hf_transformers.py (hunk 1 above)
Comment (dakshvar22) [reply rasa/nlu/utils/hugging_face/hf_transformers.py]: Ahh sorry! indeed ``TEXT`` and ``RESPONSE``

File: rasa/nlu/utils/hugging_face/transformers_pre_post_processors.py (hunk 4 above)
Comment (dakshvar22) [reply rasa/nlu/utils/hugging_face/transformers_pre_post_processors.py]: id in vocab

File: rasa/nlu/utils/hugging_face/hf_transformers.py (hunk 1 above)
Comment (dakshvar22) [reply rasa/nlu/utils/hugging_face/hf_transformers.py]: Fixed

File: rasa/nlu/utils/hugging_face/transformers_pre_post_processors.py (hunk 4 above)
Comment (tabergma) [reply rasa/nlu/utils/hugging_face/transformers_pre_post_processors.py]: minor: should we add it somewhere? it is mentioned  couple of times, maybe when we create those ids?

Comment (dakshvar22) [reply rasa/nlu/utils/hugging_face/transformers_pre_post_processors.py]: added 👍

**Pull request merged**